src/adapters/base.js

```diff
@@ -16,7 +16,25 @@ export default class BaseAdapter {
   }
 
   /**
-   * Adds a record to the IndexedDB database.
+   * Opens a connection o the database.
+   *
+   * @return {Promise}
+   */
+  open() {
+    throw new Error("Not Implemented.");
```

Comment:
Hmm actually, scratch my latest comment; nobody is actually calling `#close()` as it's a new method introduced by this patch :)

I think we can live with forcing implementers to explicitly provide their `#close()` implementation.